hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -810,7 +842,17 @@ private String intArrayToString(int[] ints) {
         }
     }
 
-    private static Column createFilterCondition(Dataset dataset,String[] allColIdInSpark, Qualifier[][] qual_list, int[] baseColumnMap, DataValueDescriptor probeValue) throws StandardException {
+    /// check that we don't access a column that's not there with baseColumnMap
+    private void checkNumColumns(String location, int[] baseColumnMap, Dataset<Row> table) throws StandardException {
+        if( baseColumnMap.length > table.schema().fields().length) {
```

Comment:
is only ">" condition is error? Should not it be "!=" here?

Reply:
> is enough. That means your File eg CSV has more columns than you specified, which is a case we can handle

Reply:
actually, you were right